src/Contacts/TSThread.m

```diff
@@ -248,9 +248,11 @@ - (NSString *)lastMessageLabel {
 - (void)updateWithLastMessage:(TSInteraction *)lastMessage transaction:(YapDatabaseReadWriteTransaction *)transaction {
     NSDate *lastMessageDate = lastMessage.date;
 
-    if ([lastMessage isKindOfClass:[TSIncomingMessage class]]) {
```

Comment:
I don't think that these several lines are determining the order of anything - rather just what's displayed in the inbox view, which shows the last message for each thread.